src/coreclr/jit/jitmetadatalist.h

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// clang-format off
+
+#ifdef JITMETADATA
+#define JITMETADATAINFO(name, type, flags) JITMETADATA(name, type, flags)
+#define JITMETADATAMETRIC(name, type, flags) JITMETADATA(name, type, flags)
+#endif
+
+#if !defined(JITMETADATAINFO) || !defined(JITMETADATAMETRIC)
+#error Define JITMETADATAINFO and JITMETADATAMETRIC before including this file.
+#endif
+
+//              Name,                          type              flags
+JITMETADATAINFO(MethodFullName,                const char*,      0)
+JITMETADATAINFO(TieringName,                   const char*,      0)
```

Comment:
What is the difference between `JITMETADATAINFO` and `JITMETADATAMETRIC`? (Document above).

Also, the answers to MethodFullName and TieringName are hard coded in SPMI playback. Should that be noted somewhere?

Reply:
Added this comment above:
```cpp
// List of metadata that the JIT can report. There are two categories:
//
// - JITMETADATAINFO: General info that can be of any type and that cannot be
//   aggregated in straightforward ways. These properties are not handled
//   automatically; the JIT must explicitly report them using
//   JitMetadata::report, and the SPMI side needs to manually handle (or ignore)
//   them in ICorJitInfo::reportMetadata.
//
// - JITMETADATAMETRIC: Metrics which are numeric types (currently int, double
//   and int64_t types supported). Their reporting is handled automatically and
//   they will be propagated all the way into SPMI replay/diff results.
```